dev/Interop/CameraCaptureUI/CameraCaptureUI/CameraCaptureUI.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#include "pch.h"
+#include <future>
+#include <ppltasks.h>
+#include <pplawait.h>
+#include <ShlObj.h>
+#include "CameraCaptureUI.h"
+#include "Microsoft.Windows.Media.Capture.CameraCaptureUI.g.cpp"
+#include "CameraCaptureUITelemetry.h"
+#include "TelemetryHelper.h"
+
+using namespace winrt::Windows::UI;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::ApplicationModel::DataTransfer;
+
+static constexpr const wchar_t* CameraAppPackageName = L"Microsoft.WindowsCamera_8wekyb3d8bbwe";
+static constexpr const wchar_t* CameraPickerUri = L"microsoft.windows.camera.picker:";
+
+
+namespace winrt::Microsoft::Windows::Media::Capture::implementation
+{
+    IAsyncOperation<hstring> LaunchCameraForResultToken(winrt::Microsoft::UI::WindowId const& targetWindow, ValueSet const& properties)
+    {
+        LauncherOptions options;
+        if (targetWindow.Value)
+        {
+            check_hresult(options.as<IInitializeWithWindow>()->Initialize(reinterpret_cast<HWND>(targetWindow.Value)));
+        }
+
+        options.TreatAsUntrusted(false);
+        options.DisplayApplicationPicker(false);
+        options.TargetApplicationPackageFamilyName(CameraAppPackageName);
+
+        Uri launchUri(CameraPickerUri);
+        winrt::Windows::System::LaunchUriResult result = co_await Launcher::LaunchUriForResultsAsync(launchUri, options, properties);
+        if (!result.Result())
+        {
+            co_return L"";
+        }
+        else if (auto t = result.Result().TryLookup(L"SelectedTokens"))
+        {
+            co_return t.as<hstring>();
+        }
+        else
+        {
+            throw hresult_canceled();
+        }
+    }
+
+    concurrency::task<token_and_path> CreateEmptyFileAndGetToken(hstring const& fileExtension)
+    {
+        auto tempFolder = ApplicationData::Current().TemporaryFolder();
+        auto tempFile = co_await tempFolder.CreateFileAsync(L"CCapture" + fileExtension, CreationCollisionOption::GenerateUniqueName);
```

Comment:
Any reasoning why the "CCapture" name is used here? I assume CCapture stands for CameraCapture. Can we use the CameraCapture sort of name instead? the reasoning behind that is it is better human readable name. 

Having clear file name would help the program to determine the source of the file as well.

Reply:
That makes sense. For better readability, I'll go ahead and use the full name instead.

Reply:
Updated!